Source/MLX/MLXArray+Indexing.swift

```diff
@@ -108,43 +115,31 @@ extension MLXArray {
         }
     }
 
-    // TODO: error checking? subscript out of range, too many indices
-
-    /// Single index subscript.
+    /// Indexing using an integer index with an axis (deprecated).
+    ///
+    /// This method supports a range expression with an axis, for example this python code:
+    ///
+    /// ```python
+    /// # python
+    /// array[..., 3]
+    /// ```
     ///
-    /// This provides access to the given index on axis 0.
+    /// This can be translated to swift as:
     ///
     /// ```swift
-    /// // print the contents of row 1
-    /// print(array[1])
+    /// array[.ellipsis, 3]
+    /// ```
+    ///
+    /// This particular subscript operator is deprecated in favor of that, but for
+    /// completeness the equivalent would be:
     ///
-    /// // set the value of row 1 to [23] (with broadcast)
-    /// array[1] = 23
+    /// ```swift
+    /// array[3, axis: -1]
     /// ```
-    public subscript(index: Int, stream stream: StreamOrDevice = .default) -> MLXArray {
-        get {
-            // see mlx_get_item_int
-            return take(resolve(index: index, axis: 0), axis: 0, stream: stream)
-        }
-        set {
-            // see mlx_set_item_int()
-
-            precondition(dtype == newValue.dtype, "\(dtype) != \(newValue.dtype)")
-
-            // trim off any singleton dimensions
-            let updateShape = Array(newValue.shape.trimmingPrefix { $0 == 1 })
-
-            // this is the shape we will set via broadcast.
-            var broadcastShape = self.shape
-            broadcastShape[0] = 1
-
-            let expanded = newValue.reshaped(updateShape).broadcast(to: broadcastShape.asInt32)
-
-            let indices = [resolve(index: index, axis: 0)]
-            self.update(scattered(indices: indices, updates: expanded, axes: [0]))
-        }
-    }
-
+    ///
+    /// ### See Also
+    /// - <doc:indexing>
+    @available(*, deprecated, message: "please use subscript(.ellipsis, 3) or equivalent")
     public subscript(index: Int, axis axis: Int, stream stream: StreamOrDevice = .default)
```

Comment:
A few of the old subscript operators are still here -- these still provide API surface that is not replaced, though the functionality is.

In a future PR I may move these into their own file just to clean this up.  I didn't do that here as it would just make this unnecessarily larger.